### PR TITLE
Enforce style

### PR DESCRIPTION
**Pull request by Baltoli**
This PR is a mechanical[^1] fix for #994; it selects some default naming conventions for C++ code and applies them across the project. Open to bikeshedding if there are particular places anyone doesn't like, but as with all these changes the most important thing is having the tool enabled at all.

The changes might not be perfect; there's a lot of surface area to cover here but they should be pretty good. I've tested against the K integration test suite to make sure nothing is broken there.

[^1]: `clang-tidy` + a pile  of one-off `sed` scripts to cover edge cases; fortunately style is easier to enforce than to apply so `clang-tidy` will do on its own moving forwards.

## Summary by CodeRabbit

- **Refactor**
  - Standardized naming conventions across the codebase for improved consistency and readability. This includes converting camelCase to snake_case and adjusting enum values to PascalCase where applicable.
  - Renamed various entities (functions, variables, classes) to adhere to the new naming convention, enhancing code clarity.
- **Style**
  - Updated comments and documentation to reflect changes in naming conventions and code structure.
- **Chores**
  - Adjusted method signatures and variable names for consistency with the newly adopted naming standards.

### CODE REVIEW

File: include/kllvm/binary/serializer.h [hunk 1]
```diff
@@ -28,10 +28,10 @@ class serializer {
...
   enum flags {
     NONE = 0,
-    DROP_HEADER = 1,
-    DROP_ARITY = 2,
-    DROP_BOTH = 3,
-    NO_INTERN = 4,
+    DropHeader = 1,
+    DropArity = 2,
+    DropBoth = 3,
+    NoIntern = 4,
```
Comment (Scott-Guest) [inline include/kllvm/binary/serializer.h:34]: Shouldn't this be `lower_case` based on  `readability-identifier-naming.EnumCase`?

File: include/kllvm/parser/KOREScanner.h [hunk 2]
```diff
@@ -29,36 +29,36 @@ enum class token {
   COLONEQUAL,
   ID,
   STRING,
-  TOKEN_EOF,
```
Comment (Scott-Guest) [inline include/kllvm/parser/KOREScanner.h:32]: All the other `token` enum values need to be updated to `CamelCase` as well

File: include/runtime/header.h [hunk 3]
```diff
@@ -25,11 +25,11 @@
 #include <runtime/collections/rangemap.h>
 #include <unordered_set>
 
-struct MatchLog {
+struct match_log {
   enum { SUCCESS = 0, FUNCTION, FAIL } kind;
```
Comment (Scott-Guest) [inline include/runtime/header.h:29]: Again, shouldn't this be `lower_case` based on `readability-identifier-naming.EnumCase`? Or `CamelCase` based on `readability-identifier-naming.EnumConstantCase`? Not sure which one applies in this case.

**Review by Scott-Guest — APPROVED** [review]
LGTM, except a few of the enums need to be fixed up.

The style is a bit different from my personal preference (I like `CamelCase` types), but I'm happy enough with the current selection.

**Comment by Scott-Guest** [comment]
Also oh my the noise from CodeRabbit is pretty bad here :grimacing: Not the biggest fan from what I've seen so far

**Comment by Baltoli** [comment]
> few of the enums need to be fixed up

Will sort these out and merge!

> Also oh my the noise from CodeRabbit is pretty bad here 😬 Not the biggest fan from what I've seen so far

Yeah, not sure I love it really either. Will give it one more go on a more "semantic" PR but I'm inclined to turn it off.

**Comment by Baltoli** [comment]
> I can probably actually just get it to PascalCase types without too much extra effort

In fact, I might do this and `using` / `typedef` in a follow-up change just so that I can get this initial large diff merged; it's becoming a bit unwieldy to work with

File: include/runtime/header.h (hunk 3 above)
Comment (Baltoli) [reply include/runtime/header.h:29]: As above, it's `EnumConstantCase` - should be `CamelCased`
